apps/server/src/modules/h5p-editor/service/contentStorage.service.ts

```diff
@@ -0,0 +1,335 @@
+import {
+	ContentId,
+	IContentMetadata,
+	IContentStorage,
+	IFileStats,
+	ILibraryName,
+	IUser as ILumiUser,
+	LibraryName,
+	Permission,
+} from '@lumieducation/h5p-server';
+import {
+	HttpException,
+	Inject,
+	Injectable,
+	InternalServerErrorException,
+	NotAcceptableException,
+	NotFoundException,
+	UnprocessableEntityException,
+} from '@nestjs/common';
+import { S3ClientAdapter } from '@shared/infra/s3-client';
+import { ErrorUtils } from '@src/core/error/utils';
+import { Readable } from 'stream';
+import { H5pFileDto } from '../controller/dto/h5p-file.dto';
+import { H5PContent } from '../entity';
+import { H5P_CONTENT_S3_CONNECTION } from '../h5p-editor.config';
+import { H5PContentRepo } from '../repo';
+import { LumiUserWithContentData } from '../types/lumi-types';
+
+@Injectable()
+export class ContentStorage implements IContentStorage {
+	constructor(
+		private readonly repo: H5PContentRepo,
+		@Inject(H5P_CONTENT_S3_CONNECTION) private readonly storageClient: S3ClientAdapter
+	) {}
+
+	private checkExtendedUserType(user: ILumiUser): void {
+		const isExtendedUserType = user instanceof LumiUserWithContentData;
+
+		if (!isExtendedUserType) {
+			throw new Error('Method expected LumiUserWithContentData instead of IUser');
+		}
+	}
+
+	private async createOrUpdateContent(
+		contentId: ContentId,
+		user: LumiUserWithContentData,
+		metadata: IContentMetadata,
+		content: unknown
+	): Promise<H5PContent> {
+		let h5pContent: H5PContent;
+
+		if (contentId) {
+			h5pContent = await this.repo.findById(contentId);
+			h5pContent.metadata = metadata;
+			h5pContent.content = content;
+		} else {
+			h5pContent = new H5PContent({
+				parentType: user.contentParentType,
+				parentId: user.contentParentId,
+				creatorId: user.id,
+				schoolId: user.schoolId,
+				metadata,
+				content,
+			});
+		}
+		return h5pContent;
+	}
+
+	public async addContent(
+		metadata: IContentMetadata,
+		content: unknown,
+		user: LumiUserWithContentData,
+		contentId?: ContentId | undefined
+	): Promise<ContentId> {
+		try {
+			this.checkExtendedUserType(user);
+
+			const h5pContent = await this.createOrUpdateContent(contentId as string, user, metadata, content);
+			await this.repo.save(h5pContent);
+
+			return h5pContent.id;
+		} catch (error) {
+			throw new HttpException('message', 500, {
+				cause: new InternalServerErrorException(error as string, 'ContentStorage:addContent'),
+			});
+		}
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	public async addFile(contentId: string, filename: string, stream: Readable, _user?: ILumiUser): Promise<void> {
```

Comment:
you can simply remove the _user, you still fulfill the interface even if you define less arguments.
And then you dont need the eslint comment.
Do the same on all occurences

Reply:
solved